backend/app.ts

```diff
@@ -1,56 +1,110 @@
 import express, { NextFunction, Request, Response } from "express";
 import path from "path";
-import cors from "cors";
-import passport from "passport";
-import { Strategy as LocalStrategy } from "passport-local";
-
-import { DBAuthenticationError } from "./error/500s";
-import { statusCodedError } from "./error/statusCodedError";
+const cors = require("cors");
 const multer = require("multer");
 const upload = multer();
+const passport = require("passport");
+const jwt = require("jsonwebtoken");
 
+const passportConfig = require("./passport");
 import { employeeRouter } from "./routes/employeeRouters";
+import { createOneEmployee } from "./controllers/employeeControllers";
+import { Employee } from ".prisma/client";
+import { EmployeeInsert } from "./interfaces/employeeInterface";
 
 // -------------------firing express app
 const app = express();
 
 //body paramter enable
 app.use(upload.array());
 
+passportConfig();
+
+app.use(passport.initialize());
 app.use(express.json());
 app.use(cors());
 app.use(express.urlencoded({ extended: false }));
 app.use(express.static(path.join(__dirname, "client/build")));
 
-passport.use(new LocalStrategy({ usernameField: "email" },
-  function (username, password, done) {
-    if (username !== "admin") {
-      return done(null, false, { message: "Username is wrong"});
-    }
-    if (password !== "password") {
-      return done(null, false, { message: "Password is wrong" });
-    }
-    return done(null, { username: "admin", message:"Login Successful" });
+// -------------------unprotected routes
+app.get("/unprotected", (request: Request, response: Response) => {
+  response.json({ msg: "unprotected" });
+});
+
+app.post("/signin", async (req, res, next) => {
+  try {
+    passport.authenticate(
+      "local",
+      (passportError: Error, user: Employee, info: any) => {
+        if (passportError || !user) {
+          res.status(400).json({ message: info.reason });
+          return;
+        }
+        req.login(user, { session: false }, (loginError) => {
+          if (loginError) {
+            res.send(loginError);
+            return;
+          }
+          const token = jwt.sign(
+            { id: user.id, email: user.email },
+            "jwt-secret-key",
+            { expiresIn: "1d" }
```

Comment:
I have put in the json request a `rememberMe: boolean` variable, as we were discussing we could give a longer expiration (ex: 7 days, 30 days) where `rememberMe` is true and a short expiration (1 hr, 2 hr) where it is false

Reply:
https://github.com/Free-Laborers/u-refer/blob/43f26c426cca69c38bcdec3f825760ff90662e45/frontend/src/pages/Login/Login.tsx#L59-L63 
https://github.com/Free-Laborers/u-refer/blob/43f26c426cca69c38bcdec3f825760ff90662e45/frontend/src/pages/Login/Login.tsx#L69-L75